midend/actionSynthesis.cpp

```diff
@@ -135,23 +124,17 @@ const IR::Node* DoSynthesizeActions::preorder(IR::P4Control* control) {
 }
 
 const IR::Node* DoSynthesizeActions::postorder(IR::P4Control* control) {
-    if (actions.empty())
-        return control;
-    auto nm = new IR::IndexedVector<IR::Declaration>(*control->controlLocals);
     for (auto a : actions)
-        nm->push_back(a);
-    auto result = new IR::P4Control(control->srcInfo, control->name,
-                                    control->type, control->constructorParams,
-                                    nm, control->body);
-    return result;
+        control->controlLocals.push_back(a);
```

Comment:
This is one thing that I found would be nice when debugging: if you mutate a Node, allocate a new ID.
This has a very nice property: the object ID tells you exactly when it was created, so it is much easier to put breakpoints to figure out when a specific node was inserted in the graph.
Maybe I should file an 'enhancement' issue with the compiler.